File: Golang/Set Mismatch.go
func findErrorNums(nums []int) []int {
    var sum, dup int
    n:= len(nums)

    // compute the sum and find the duplicate
    // note that the elements of nums are +ve
    foundDuplicate := false
    for _, x := range nums {
        x1 := abs(x)
        sum += x1
        if !foundDuplicate {
            if nums[x1-1] < 0 {
                dup = x1
                foundDuplicate = true
            } else {
                nums[x1-1] = -nums[x1-1]
            }
        }
    }

    // assemble dup and missing
    sumExpected, sumUniq := n*(n+1)/2, sum - dup
    return []int{dup, sumExpected - sumUniq}
}

func abs(a int) int {
    if a < 0 {
        return -a
    }
    return a
}

File: Python3/Set Mismatch.py
#1
class Solution:
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        total = len(nums)*(1+len(nums)) // 2
        miss = total - sum(set(nums)) 
        rep = sum(nums) - sum(set(nums)) 

        ans = []
        ans.append(rep)
        ans.append(miss)
        
        return ans

#2
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        m = {}
        ans = []
        for i in range(1, len(nums)+1):
            if nums[i-1] in m:
                m[nums[i-1]] = m[nums[i-1]] + 1
                if m[nums[i-1]] == 2:
                    ans.append(nums[i-1]) 
            else:
                m[nums[i-1]] = 1
        
        for i in range(1,  len(nums) + 1):
            if i not in m:
                ans.append(i)

        return ans